simulationworkflowschema/general.py

```diff
@@ -231,24 +285,213 @@ def normalize(self, archive, logger):
+                    )
                 elif len(self._calculations) == len(self._systems):
-                    inputs.append(Link(name=INPUT_SYSTEM_NAME, section=self.input_structure if n == 0 else self._systems[n - 1]))
-                    outputs.append(Link(name=OUTPUT_SYSTEM_NAME, section=self._systems[n]))
+                    inputs.append(
+                        Link(
+                            name=INPUT_SYSTEM_NAME,
+                            section=self.input_structure
+                            if n == 0
+                            else self._systems[n - 1],
+                        )
+                    )
+                    outputs.append(
+                        Link(name=OUTPUT_SYSTEM_NAME, section=self._systems[n])
+                    )
                 else:
                     continue
                 if calculation.method_ref:
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=calculation.method_ref))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=calculation.method_ref)
+                    )
                 elif len(self._calculations) == len(self._methods):
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=self._methods[n]))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=self._methods[n])
+                    )
                 elif len(self._methods) == 1:
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=self._methods[0]))
-                self.tasks.append(Task(name=f'Step {n}', inputs=inputs, outputs=outputs))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=self._methods[0])
+                    )
+                self.tasks.append(
+                    Task(name=f"Step {n}", inputs=inputs, outputs=outputs)
+                )
+
+
+class BeyondDFT(SerialSimulation):
+    """
+    Base class used to normalize standard workflows beyond DFT containing two specific
+    SinglePoint tasks (GWWorkflow = DFT + GW, DMFTWorkflow = DFT + DMFT,
+    MaxEntWorkflow = DMFT + MaxEnt, and so on) and store the outputs in the self.results
+    section.
+    """
+
+    def _resolve_outputs_section(self, output_section, task: TaskReference) -> None:
+        """
+        Resolves the output_section of a task and stores the results in the output_section.
+
+        Args:
+            task (TaskReference): The task from which the outputs are got.
+        """
+        for name, section in output_section.m_def.all_quantities.items():
+            name = f"{name}_electronic" if name in ["dos", "band_structure"] else name
+            try:
+                calc_section = getattr(task.outputs[-1].section, name)
+                if calc_section:
+                    output_section.m_set(section, calc_section)
+            except Exception:
+                continue
+
+    def get_gw_workflow_results(
```

Comment:
I think all functions and variables should be private so as not to clutter the quantites namespace

Reply:
I'd like to keep it like this, but maybe returning results so anybody can use this methods for resolving task outpus?

Reply:
i think all the get_*_workflow_results functions can be generalized to get_worklfow_results. simply have self.results[f'{method}_outputs'] = outputs

Reply:
True, I generalized it now.